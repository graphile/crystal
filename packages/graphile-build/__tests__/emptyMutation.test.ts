const { buildSchema, defaultPlugins } = require("../");

test("generates empty schema (with no Mutation type)", async () => {
  const schema = await buildSchema([...defaultPlugins]);
  expect(schema).toMatchSnapshot();
});
