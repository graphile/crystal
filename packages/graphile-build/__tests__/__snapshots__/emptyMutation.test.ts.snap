// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates empty schema (with no Mutation type) 1`] = `
"""An object with a globally unique \`ID\`."""
interface Node {
  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  id: ID!
}

"""The root query type which gives access points into the data universe."""
type Query implements Node {
  """
  Exposes the root query type nested one level down. This is helpful for Relay 1
  which can only query top level fields if they are in a particular form.
  """
  query: Query!

  """
  The root query type must be a \`Node\` to work well with Relay 1 mutations. This just resolves to \`query\`.
  """
  id: ID!

  """Fetches an object given its globally unique \`ID\`."""
  node(
    """The globally unique \`ID\`."""
    id: ID!
  ): Node
}

`;
